src/Promotion/Processor/AfterTaxPromotionProcessor.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Advanced Promotion plugin for Sylius.
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusAdvancedPromotionPlugin\Promotion\Processor;
+
+use MonsieurBiz\SyliusAdvancedPromotionPlugin\Context\PromotionContextInterface;
+use MonsieurBiz\SyliusAdvancedPromotionPlugin\Entity\AfterTaxAwareInterface;
+use Sylius\Component\Promotion\Action\PromotionApplicatorInterface;
+use Sylius\Component\Promotion\Checker\Eligibility\PromotionEligibilityCheckerInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Sylius\Component\Promotion\Processor\PromotionProcessorInterface;
+use Sylius\Component\Promotion\Provider\PreQualifiedPromotionsProviderInterface;
+use Webmozart\Assert\Assert;
+
+final class AfterTaxPromotionProcessor implements PromotionProcessorInterface
+{
+    public function __construct(
+        private PreQualifiedPromotionsProviderInterface $preQualifiedPromotionsProvider,
+        private PromotionEligibilityCheckerInterface $promotionEligibilityChecker,
+        private PromotionApplicatorInterface $promotionApplicator,
+        private PromotionContextInterface $promotionContext
+    ) {
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function process(PromotionSubjectInterface $subject): void
+    {
+        // Set the promotion context to after cart
+        $this->promotionContext->setAfterTax(true);
+
+        /**
+         * We do not revert applied promotions because we want to keep the rules applies before tax.
+         *
+         * @see \Sylius\Component\Promotion\Processor\PromotionProcessor::process()
+         */
+
+        // Retrieve promotion after tax, it's a double security because the PromotionTaxContextEligibilityChecker si also called
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏ typo in the comment
```suggestion
        // Retrieve promotion after tax, it's a double security because the PromotionTaxContextEligibilityChecker is also called
```